netmiko/ssh_autodetect.py

```diff
@@ -150,6 +151,12 @@
         "priority": 99,
         "dispatch": "_autodetect_std",
     },
+    "brocade": {
+        "cmd": "show version",
+        "search_patterns": [r"Brocade Communications Systems"],
+        "priority": 99,
+        "dispatch": "_autodetect_std",
+    },
```

Comment:
I am happy to help though I have no idea what the correct identification string is for each of these platforms...though maybe @LindsayHill knows?

Reply:
I can probably get some. I doubt there is a master list. They will have been changing with recent software releases too, referring to the new owner.

Reply:
Here's an example of a box im trying to do:
```
SSH@RMPK232254-1#show ver
System Mode: MLX
Chassis: NetIron 16-slot (Serial #: SA11111111,  Part #: 35522-000A)
NI-X-SF  Switch Fabric Module 1 (Serial #: SA11111111,  Part #: 35523-300C)
FE 1: Type fe200,  Version 2
FE 2: Type fe200,  Version 2
FE 3: Type fe200,  Version 2
NI-X-SF  Switch Fabric Module 2 (Serial #: SA11111111,  Part #: 35523-300C)
FE 1: Type fe200,  Version 2
FE 2: Type fe200,  Version 2
FE 3: Type fe200,  Version 2
NI-X-SF  Switch Fabric Module 3 (Serial #: SA11111111,  Part #: 35523-300C)
FE 1: Type fe200,  Version 2
FE 2: Type fe200,  Version 2
FE 3: Type fe200,  Version 2
==========================================================================
SL M1: NI-MLX-MR Management Module Active (Serial #: SA11111111, Part #: 35524-102A):
Boot     : Version 5.1.0aT165 Copyright (c) 1996-2009 Brocade Communications Systems, Inc.
Compiled on Jan  8 2011 at 07:35:30 labeled as xmprm05100a
 (524006 bytes) from boot flash
Monitor  : Version 5.1.0aT165 Copyright (c) 1996-2009 Brocade Communications Systems, Inc.
Compiled on Jan  8 2011 at 07:35:00 labeled as xmb05100a
 (524001 bytes) from code flash
IronWare : Version 5.1.0cT163 Copyright (c) 1996-2009 Brocade Communications Systems, Inc.
Compiled on Apr  8 2011 at 21:10:48 labeled as xmr05100c
 (7001256 bytes) from Primary
Board ID : 00 MBRIDGE Revision : 32
916 MHz Power PC processor 7447A (version 8003/0101) 166 MHz bus
512 KB Boot Flash (AM29LV040B), 32 MB Code Flash (MT28F128J3)
1024 MB DRAM
Active Management uptime is 1042 days 9 hours 22 minutes 2 seconds```

Reply:
We can just use `NetIron` as the search pattern for that one and set it to `brocade_netiron` for the device type.

Reply:
VDX switches run "NOS" (Network Operating System). So `brocade_nos` & `brocade_vdx` are the same thing.

SLX devices run SLX-OS, which is an evolution of NOS, with some of the MPLS & routing capabilities from NetIron added in.

I believe netmiko refers to it as "ExtremeSlxSSH" (has been a while since I looked through netmiko code).